### PR TITLE
#60 Add Map block component

### PR DESCRIPTION
**Pull request by katarinazaprazna**
- Export new OLO icons
- Create a placeholder for the map functionality
- Create the component
- NOTE: Bold text was not used on purpose (see Figma)

### CODE REVIEW

File: next/src/components/common/Box/Map.tsx [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        className,
+      )}
+    >
+      <MapPlaceholder />
+
+      <div className="flex flex-col items-center justify-center px-4 lg:px-6 lg:py-3">
+        {children.map((child: ReactNode, index) => {
+          return (
+            // eslint-disable-next-line react/no-array-index-key
+            <div key={index} className="w-full">
```
Comment (radoslavzeman) [inline next/src/components/common/Box/Map.tsx:32]: Do we need an additional `div` here? Could we use `Fragment` instead? :)

File: next/src/components/common/Box/Map.tsx [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  children: ReactNode[]
+  className?: string
+}
+
+/**
+ *   Figma: https://www.figma.com/design/2qF09hDT9QNcpdztVMNAY4/OLO-Web?node-id=1341-10975&m=dev
+ */
+
+const Map = ({ children, className }: MapProps) => {
+  if (!children?.length) return null
```
Comment (radoslavzeman) [inline next/src/components/common/Box/Map.tsx:17]: I would prefer to render the component also with no children (empty right part). It's more understandable for content admins if they see "bad looking" component than no component.

File: next/src/components/common/MapPlaceholder.tsx [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+import mapImage from '@/src/assets/images/olo-placeholder-map.jpg'
+
+const MapPlaceholder = () => {
+  return (
+    <Image
+      src={mapImage}
+      alt="Mapa"
+      width={324}
+      height={220}
```
Comment (radoslavzeman) [inline next/src/components/common/MapPlaceholder.tsx:11]: The map should adapt it's height depending on the content in the right part (rows), it should not have fixed dimensions.
The map can have fixed width.

**Review by radoslavzeman — CHANGES_REQUESTED** [review]

**Review by radoslavzeman — COMMENTED** [review]
Nice and clean, please only reflect minor comments left in the review :)

File: next/src/components/common/Box/Map.tsx (hunk 1 above)
Comment (katarinazaprazna) [reply next/src/components/common/Box/Map.tsx:32]: We're using the `className` property on each of the divs present in the code, which, to my knowledge, makes the usage of fragments impossible

File: next/src/components/common/Box/Map.tsx (hunk 2 above)
Comment (katarinazaprazna) [reply next/src/components/common/Box/Map.tsx:17]: This is the pattern of rendering we're currently using in the rest of the code. In this case, our options would be 1. defaulting to rendering an empty rectangle or 2. rendering a map without markers. Is this something we want to consider doing?

File: next/src/components/common/MapPlaceholder.tsx (hunk 3 above)
Comment (katarinazaprazna) [reply next/src/components/common/MapPlaceholder.tsx:11]: This was something I planned to revisit once implementing the actual, Mapbox mapping functionality. The placeholder is temporary

File: next/src/components/common/Box/Map.tsx (hunk 1 above)
Comment (radoslavzeman) [reply next/src/components/common/Box/Map.tsx:32]: If you remove `items-center` from parent div, you can use `Fragment` :)

**Review by radoslavzeman — APPROVED** [review]

**Pull request merged**